packages/nx/src/config/project-graph.ts

```diff
@@ -114,15 +120,21 @@ export interface ProjectGraphProjectNode {
  *
  */
 export interface ProjectGraphExternalNode {
-  type: 'npm';
-  name: `npm:${string}`;
+  type: 'npm' | 'cargo' | 'gradle';
```

Comment:
Similar to the `name` just being a string, nx-dotnet would add external nodes that had type `nuget` or similar so this should also probably just be a string.